### PR TITLE
Add gedi_subsetter H5DataFrame and h5coro methods to template notebook

### PR DESCRIPTION
**Pull request by weiji14**
Main change is adding the H5DataFrame read in method from #5 into the `benchmark-format-template.ipynb` notebook. Also pulling in the h5coro method from the `benchmark-original-hdf5.ipynb` notebook at c9ac2100e38253334dba2f0e1b7d5a359a6be8d0, and added tqdm progress bars.

### CODE REVIEW

File: notebooks/benchmark-format-template.ipynb [hunk 1]
```diff
@@ -163,7 +165,9 @@
    },
    "outputs": [],
    "source": [
-    "final_array"
+    "%%time\n",
+    "assert final_array.shape == (298271231,)\n",
+    "final_array.mean()"
```
Comment (weiji14) [inline notebooks/benchmark-format-template.ipynb]: @abarciauskas-bgse, were you thinking of concatenating the `h_ph` (photon height) data variable across all 5 ATL03 files into a single 1D array just like xarray? Or some other shape?

Comment (weiji14) [reply notebooks/benchmark-format-template.ipynb]: Ah ok, I see that your new commit at c9ac2100e38253334dba2f0e1b7d5a359a6be8d0, let me reconcile those together later (after lunch).

Comment (abarciauskas-bgse) [reply notebooks/benchmark-format-template.ipynb]: Thanks @weiji14 I was just about to comment about that!

Comment (weiji14) [reply notebooks/benchmark-format-template.ipynb]: Ok, have synced up with the stuff from commit e9aedc4905a85d2349a3c272cf4f6b8399089877, going to squash merge this into main now!

**Pull request merged**